### DISCUSSION TITLE
add option of `caseSensitive`(true or false) when validate plugin test scenario data

### DISCUSSION THREAD

**Discussion by drgnchan** [apache/skywalking#10204] [Ideas]
Within work directory of skywalking-java,I run `./test/plugin/run.sh -f jedis-4.x-scenario`.Then I found some validation error in Terminal:

<img width="690" alt="image" src="https://user-images.githubusercontent.com/40224023/209503856-b23b941d-c31b-4fd3-9c21-48d5c75f0e51.png">

part of validation:

```
actual: 
  Segment[4cec05f8e5424f75ad3e771f548beddb.37.16720286701050000] e
  expected:     Span[0, 2] Jedis/set
  actual:       span[0, 2] Jedis/SET
  reason:       [operation name]: expected=>{Jedis/set}, actual=>{Jedis/SET}
```

```
actual: 
  Segment[4ff9b11162b745a3a3d5bc28c9e34038.39.16720285327380000] e
  expected:     Span[0, 1] Jedis/ECHO
  actual:       span[0, 1] Jedis/ECHO
  reason:       [tag(cache.cmd) value]: expected=>{echo}, actual=>{ECHO}
```

And I think case-insensitive validation(if so,this scenario will succeed) is reasonable in this scenario.I propose that we make a place for every scenario to set caseSensitive true or false in `expectedData.yaml`.And I can do this job if the proposal has no problem.

**Comment by wu-sheng** [comment]
Why can't you just simply correct the expected-data file?

**Reply by drgnchan** [reply]
Sometimes operation name is generated dynamically.If a dependency change it's field which forms operation name from upper-case to lower-case,the plugin test scenario expected data should change accordingly because there is no case-insensitive validation.

**Comment by wu-sheng** [comment]
And all tests are passed in every PR/commit. How could this one fail on your side? Maybe you are using a different client version?

**Reply by drgnchan** [reply]
> And all tests are passed in every PR/commit. How could this one fail on your side? Maybe you are using a different client version?

I just run `./test/plugin/run.sh -f jedis-4.x-scenario` local.And I found that there is no `-f` option in `action.yml` bash step which is a bit different from my command.

https://github.com/apache/skywalking-java/blob/cd84ba2167fd2ca7ccd8531b035d390fadd582a9/.github/actions/run/action.yml#L54

**Reply by wu-sheng** [reply]
So? `-f` is just a script parameter. I don't think these two are relative.

**Reply by drgnchan** [reply]
> So? `-f` is just a script parameter. I don't think these two are relative.

Yes.There is nothing to do with `-f`(Actually I make `-f` in my own action to check).I am trying to figure out why the actualData.yaml showed `Jedis/ECHO` in local but 'Jedis/echo' in github action.

**Reply by wu-sheng** [reply]
The value of this tag is returned by the JDK method. 

https://github.com/apache/skywalking-java/blob/cd84ba2167fd2ca7ccd8531b035d390fadd582a9/apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-2.x-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v3/JedisMethodInterceptor.java#L44

So, if I am going to guess, this may have some relationship with your JDK vendor/version?

**Reply by drgnchan** [reply]
> The value of this tag is returned by the JDK method.
> 
> https://github.com/apache/skywalking-java/blob/cd84ba2167fd2ca7ccd8531b035d390fadd582a9/apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-2.x-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v3/JedisMethodInterceptor.java#L44
> 
> So, if I am going to guess, this may have some relationship with your JDK vendor/version?

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/40224023/209543247-95e9e2f2-2af1-48de-8bc4-77017f23fe32.png">

As the picture shows,`protocolCommand.toLowerCase()` didn't work.I am struggling with that...

**Reply by drgnchan** [reply]
The reason is that the directory`skywalking-agent` is built in Sep and its version is obviously older(no `toLowerCase()` invoking) than the latest.I am sorry to bother you. @wu-sheng
